### PR TITLE
ens lastUpdatedIndex and name wrapped event

### PR DESCRIPTION
**Pull request by rahul7668gupta**
Screenshot 2023-05-02 at 6 39 07 PM

### CODE REVIEW

File: ens/src/utils.ts [hunk 1]
```diff
@@ -134,4 +136,35 @@ export function byteArrayFromHex(s: string): ByteArray {
 export function uint256ToByteArray(i: BigInt): ByteArray {
   let hex = i.toHex().slice(2).padStart(64, '0')
   return byteArrayFromHex(hex)
+}
+
+export function decodeName(buf: Bytes, txHash: string): Array<string> | null {
```
Comment (sharathkrml) [inline ens/src/utils.ts:141]: Suggesion: can we have separate test case for utils functions?,this function seems major

**Review by sharathkrml — COMMENTED** [review]
Initial review done,Just added a small suggestion

File: ens/src/utils.ts (hunk 1 above)
Comment (rahul7668gupta) [reply ens/src/utils.ts:141]: added test for decode name

File: airstack-modules/graphql/airstack-domain-name-schema.graphql.ts [hunk 2]
```diff
@@ -71,6 +72,7 @@ type AirDomainTransferTransaction implements AirDomainEvent @entity {
   tokenId: String # dec(labelhash)  # - NA
   domain: AirDomain!
   index: BigInt! # - NA
+  lastUpdatedIndex: BigInt!
```
Comment (0xsarvesh) [inline airstack-modules/graphql/airstack-domain-name-schema.graphql.ts:75]: This field is only required in entity which will be updated. So it's need for Domain Entity only. Please remove it. 

Also refer lens schema for details

File: ens/src/name-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const chainId = getChainId();
+  let airBlock = getOrCreateAirBlock(chainId, event.block.number, event.block.hash.toHexString(), event.block.timestamp);
+  airBlock.save();
+  const domain = airstack.domain.getOrCreateAirDomain(new airstack.domain.Domain(event.params.node.toHexString(), chainId, airBlock, TOKEN_ADDRESS_ENS));
+  if (!domain.labelName && label) {
+    domain.labelName = label;
+    domain.name = name;
+    domain.lastUpdatedIndex = updateAirEntityCounter(AIR_DOMAIN_LAST_UPDATED_INDEX_ENTITY_COUNTER_ID, airBlock);
+    domain.lastUpdatedBlock = airBlock.id;
+    domain.save();
```
Comment (0xsarvesh) [inline ens/src/name-wrapper.ts:27]: Also, can you create a function to save Domains entity. That function will wrap below three statements. 
```
domain.lastUpdatedIndex = updateAirEntityCounter(AIR_DOMAIN_LAST_UPDATED_INDEX_ENTITY_COUNTER_ID, airBlock);
    domain.lastUpdatedBlock = airBlock.id;
    domain.save();

```

File: ens/tests/utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { assert, describe, test } from "matchstick-as";
+import { Bytes } from "@graphprotocol/graph-ts";
+import { decodeName } from "../src/utils";
+
+describe("Unit tests for util functions", () => {
+  test("test decodeName", () => {
```
Comment (0xsarvesh) [inline ens/tests/utils.test.ts:6]: Please add more case to this test.

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]
Please look into comments

File: ens/src/utils.ts (hunk 1 above)
Comment (rahul7668gupta) [reply ens/src/utils.ts:141]: done

File: ens/src/name-wrapper.ts (hunk 3 above)
Comment (rahul7668gupta) [reply ens/src/name-wrapper.ts:27]: done

File: ens/tests/utils.test.ts (hunk 4 above)
Comment (rahul7668gupta) [reply ens/tests/utils.test.ts:6]: done

**Review by 0xsarvesh — APPROVED** [review]

**Review by 0xsarvesh — APPROVED** [review]

**Pull request merged**